src/Model/BSONIterator.php

```diff
@@ -36,23 +36,18 @@ class BSONIterator implements Iterator
 {
     private const BSON_SIZE = 4;
 
-    /** @var string */
-    private $buffer;
+    private string $buffer;
 
-    /** @var integer */
-    private $bufferLength;
+    private int $bufferLength;
 
-    /** @var mixed */
-    private $current;
+    /** @var array|object|null */
+    private $current = null;
```

Comment:
Note that rector keeps changing this to `private ?object $current = null;`, which is incorrect as the return type of `toPHP` depends on the type map provided. @GromNaN is there a way to tell rector about this?

Reply:
It looks like we are using rector with an outdated version of phpstan stubs. https://github.com/phpstan/phpstan-src/blob/1.10.21/resources/functionMap.php#L6735

Reply:
To disable the problematic rule, add this to `rector.php`:
```php
    $rectorConfig->skip([
        // ...
        TypedPropertyFromAssignsRector::class => [
            __DIR__ . '/src/Model/BSONIterator.php',
        ],
```